test/cpp/jit/test_schema_matching.cpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <ATen/test/test_assert.h>
+#include <torch/csrc/jit/ir.h>
+#include <torch/csrc/jit/irparser.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/testing/file_check.h>
+#include <torch/jit.h>
+#include "test/cpp/jit/test_base.h"
+#include "torch/csrc/jit/custom_operator.h"
+
+#include <sstream>
+#include <string>
+
+namespace torch {
+namespace jit {
+
+void testSchemaMatching() {
+  auto run = [](std::shared_ptr<script::CompilationUnit> cu,
```

Comment:
Can we avoid using low-level APIs like this in tests, it adds work to any refactoring. Please use Module and its define methods and ways of running methods.

Reply:
actually, why are these C++ tests at all? If this is just running code but using a custom op, then just define the testing op in  register_builtin_ops and write python tests.

Reply:
I wrote the tests in c++ so that the op only gets defined when the tests are run and we don't have unnecessary ops in in the runtime.